src/Sarif.Multitool/ConvertCommand.cs

```diff
@@ -3,17 +3,41 @@
 
 using System;
 using System.Diagnostics;
+using System.Globalization;
 using Microsoft.CodeAnalysis.Sarif.Converters;
+using Microsoft.CodeAnalysis.Sarif.Driver;
 using Microsoft.CodeAnalysis.Sarif.Writers;
 
 namespace Microsoft.CodeAnalysis.Sarif.Multitool
 {
     internal static class ConvertCommand
     {
-        public static int Run(ConvertOptions convertOptions)
+        public static int Run(ConvertOptions convertOptions, IFileSystem fileSystem = null)
         {
+            if (fileSystem == null) { fileSystem = new FileSystem(); }
+
             try
             {
+                if (string.IsNullOrEmpty(convertOptions.OutputFilePath))
+                {
+                    convertOptions.OutputFilePath = convertOptions.InputFilePath + ".sarif";
+                }
+
+                if (fileSystem.DirectoryExists(convertOptions.OutputFilePath))
```

Comment:
Should this check be in ReportWhetherOutputFileCanBeCreated? Or is it too distinct from the file existence check?

Reply:
*Sigh* Darn, you're good. I think it does belong there. And note that as the code stands, `convert` is the only command that performs this check, yet it's equally applicable to all commands. I found it awkward to group the two checks together -- let me go back now and see if I can manage it.

---
In reply to: [312260058](https://github.com/microsoft/sarif-sdk/pull/1625#discussion_r312260058) [](ancestors = 312260058)

Reply:
No, I take it back. Ideally you're right but I've reached my limit on overall Multitool command cleanup when the bug being fixed is "Please respect the `--force` option."

I filed [https://github.com/microsoft/sarif-sdk/issues/1631](https://github.com/microsoft/sarif-sdk/issues/1631), "All multitool commands should do the 'directory exists' check like 'convert' does.".

---
In reply to: [312265617](https://github.com/microsoft/sarif-sdk/pull/1625#discussion_r312265617) [](ancestors = 312265617,312260058)